compositor_render/src/transformations/builtin/transformations.rs

```diff
@@ -1,57 +1,95 @@
 use std::sync::Arc;
 
 use compositor_common::scene::{
-    builtin_transformations::{BuiltinTransformation, TransformToResolution},
-    ShaderParam,
+    builtin_transformations::{BuiltinTransformationSpec, TransformToResolution},
+    Resolution, ShaderParam,
 };
 
 use crate::{
     renderer::{WgpuCtx, WgpuError},
     transformations::shader::Shader,
+    utils::rgba_to_wgpu_color,
 };
 
 pub struct BuiltinTransformations {
     transform_resolution: ConvertResolutionTransformations,
+    fixed_position_layout: FixedPositionLayout,
 }
 
 impl BuiltinTransformations {
     pub fn new(wgpu_ctx: &Arc<WgpuCtx>) -> Result<Self, WgpuError> {
         Ok(Self {
             transform_resolution: ConvertResolutionTransformations::new(wgpu_ctx)?,
+            fixed_position_layout: FixedPositionLayout::new(wgpu_ctx)?,
         })
     }
 
-    pub fn shader(&self, transformation: &BuiltinTransformation) -> Arc<Shader> {
+    pub fn shader(&self, transformation: &BuiltinTransformationSpec) -> Arc<Shader> {
         match transformation {
-            BuiltinTransformation::TransformToResolution(TransformToResolution::Stretch) => {
+            BuiltinTransformationSpec::TransformToResolution(TransformToResolution::Stretch) => {
                 self.transform_resolution.stretch.clone()
             }
-            BuiltinTransformation::TransformToResolution(TransformToResolution::Fill) => {
+            BuiltinTransformationSpec::TransformToResolution(TransformToResolution::Fill) => {
                 self.transform_resolution.fill.clone()
             }
-            BuiltinTransformation::TransformToResolution(TransformToResolution::Fit(_)) => {
+            BuiltinTransformationSpec::TransformToResolution(TransformToResolution::Fit(_)) => {
                 self.transform_resolution.fit.clone()
             }
+            BuiltinTransformationSpec::FixedPositionLayout { .. } => {
+                self.fixed_position_layout.0.clone()
+            }
         }
     }
 
-    pub fn params(transformation: &BuiltinTransformation) -> Option<ShaderParam> {
+    pub fn params(
+        transformation: &BuiltinTransformationSpec,
+        output_resolution: &Resolution,
+    ) -> Option<ShaderParam> {
         match transformation {
-            BuiltinTransformation::TransformToResolution(_) => None,
+            BuiltinTransformationSpec::TransformToResolution(_) => None,
+            BuiltinTransformationSpec::FixedPositionLayout {
+                texture_layouts: textures_layouts,
+                ..
+            } => {
+                let layouts: Vec<ShaderParam> = textures_layouts
+                    .iter()
+                    .map(|layout| {
+                        ShaderParam::Struct(vec![
+                            (
+                                "top",
+                                ShaderParam::I32(
+                                    layout.top.pixels(output_resolution.height as u32),
```

Comment:
if you extract width and hight to variable it would avoid a lot of indentation (up to you)